opendevin/runtime/runtime.py

```diff
@@ -75,32 +94,38 @@ async def run_action(self, action: Action) -> Observation:
         """
         if not action.runnable:
             return NullObservation('')
-        action_id = action.action  # type: ignore[attr-defined]
-        if action_id not in ACTION_TYPE_TO_CLASS:
-            return ErrorObservation(f'Action {action_id} does not exist.')
-        if not hasattr(self, action_id):
+        action_type = action.action  # type: ignore[attr-defined]
+        if action_type not in ACTION_TYPE_TO_CLASS:
+            return ErrorObservation(f'Action {action_type} does not exist.')
+        if not hasattr(self, action_type):
             return ErrorObservation(
-                f'Action {action_id} is not supported in the current runtime.'
+                f'Action {action_type} is not supported in the current runtime.'
             )
-        observation = await getattr(self, action_id)(action)
+        observation = await getattr(self, action_type)(action)
+        observation._parent = action.id  # type: ignore[attr-defined]
         return observation
 
-    def get_background_obs(self) -> list[CmdOutputObservation]:
+    async def _start_background_observation_loop(self):
+        while True:
+            await self.submit_background_obs()
+            await asyncio.sleep(1)
```

Comment:
Ditto, should we sleep for this long?

Reply:
This one probably yes--these are background commands accumulating logs.

If we sleep very quickly, we might get 10 observations with only a few characters each.